File: backend/src/aithru_agent/trace/spans.py
from typing import Literal

from aithru_agent.domain.base import AithruBaseModel


AgentTraceSpanKind = Literal[
    "run",
    "message",
    "todo",
    "model",
    "tool",
    "approval",
    "subagent",
    "workspace",
    "artifact",
    "external_run",
    "sandbox",
    "memory",
]

AgentTraceSpanStatus = Literal["running", "completed", "failed", "cancelled"]


class AgentTraceSpan(AithruBaseModel):
    id: str
    run_id: str
    kind: AgentTraceSpanKind
    name: str
    status: AgentTraceSpanStatus
    start_sequence: int
    started_at: str
    end_sequence: int | None = None
    ended_at: str | None = None
    refs: dict | None = None

File: backend/packages/trace/src/spans.ts
export interface AgentTraceSpan {
  id: string;
  run_id: string;
  parent_id?: string;
  kind: string;           // "run" | "message" | "tool" | "approval"
  name: string;
  started_at: string;
  completed_at?: string;
  status: "ok" | "error" | "cancelled";
  metadata?: Record<string, unknown>;
  children: AgentTraceSpan[];
}
